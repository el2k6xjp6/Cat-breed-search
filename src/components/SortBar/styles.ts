import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
  height: 60px;
  margin: 10px;
  display: flex;
  align-items: center;
  @media only screen and (max-width: 480px) {
    margin: 0;
  }
`;
