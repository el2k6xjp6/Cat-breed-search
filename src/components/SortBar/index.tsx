import DropDown from "../DropDown";
import { SORT_METHODS, SORT_METHOD_OPTIONS } from "../../constants";
import { Container } from "./styles";

interface Props {
  selection: number;
  handleClick: (value: number) => void;
}

function SortBar(props: Props) {
  return (
    <Container>
      <DropDown
        selectedText={SORT_METHODS[props.selection]}
        placeHolder="Sort By"
        options={SORT_METHOD_OPTIONS}
        handleOptionChange={props.handleClick}
      />
    </Container>
  );
}

export default SortBar;
