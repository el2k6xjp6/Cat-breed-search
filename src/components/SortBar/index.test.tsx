import SortBar from ".";
import "jest-styled-components";
import renderer from "react-test-renderer";

describe("App", () => {
  test("snapshot", () => {
    const mockProps = {
      selection: -1,
      handleClick: jest.fn,
    };
    const tree = renderer.create(<SortBar {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
