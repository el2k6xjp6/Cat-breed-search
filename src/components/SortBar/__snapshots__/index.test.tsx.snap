// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
.c1 {
  cursor: pointer;
  width: 200px;
  height: 40px;
}

.c2 {
  position: relative;
  border-radius: 4px;
  border: solid 1px #c4c4c4;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
}

.c2:hover {
  border: solid 1px black;
}

.c3 {
  text-indent: 15px;
}

.c4 {
  color: #aaaaaa;
}

.c5 {
  position: absolute;
  right: 10px;
  color: #909090;
  -webkit-transform: rotate(360deg);
  -ms-transform: rotate(360deg);
  transform: rotate(360deg);
  -webkit-transition: -webkit-transform .2s ease-in-out;
  -webkit-transition: transform .2s ease-in-out;
  transition: transform .2s ease-in-out;
}

.c0 {
  width: 100%;
  height: 60px;
  margin: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      id="DropDownHeader"
      onClick={[Function]}
    >
      <div
        className="c3 c4"
      >
        Sort By
      </div>
      <div
        className="c5"
        id="DropDownArrow"
      >
        â–¼
      </div>
    </div>
  </div>
</div>
`;
